### ISSUE TITLE
Code Compatibility Issues in the Tutorials 

### ISSUE THREAD

**Issue by AishwaryaSeth** [neurodata/ProgLearn#483]
This issue documents current code compatibility and missing package issues in the tutorials. 

#### Snapshots of documentation error:

1. ConvRF_Kappas.ipynb: No module named 'torch'. I downloaded and installed Pytorch and torchvision, and rebuilt the proglearn package, but the error still appears. 

<img width="1792" alt="ConvRF" src="https://user-images.githubusercontent.com/40052470/133530012-325decd4-8583-4370-9b10-e46ba7888fc3.png">

2. ConvRF_Kappas_3_Class.ipynb: Same issue as in 1. 

<img width="1792" alt="ConvRF_3" src="https://user-images.githubusercontent.com/40052470/133530030-fbe5254e-f48d-4dc4-9834-1276813db2b0.png">

3. fte_bte_dtd.ipynb: Mismatch in the input size. I may be missing a pre-processing step, but not sure what it was. 

<img width="1792" alt="fte_bte_dtd" src="https://user-images.githubusercontent.com/40052470/133530133-0eb3256c-0093-4bbc-9379-eff8ea0091a7.png">

4. fte_bte_flowers.ipynb: Error 1 was that the matrix for image labels has to be explicitly downloaded, and this is not mentioned in the Tutorial. Error 2 was again an input mismatch, in which I am not sure what preprocessing is to be applied. 

<img width="1792" alt="fte_bte_flowers" src="https://user-images.githubusercontent.com/40052470/133530469-8f630d97-8337-4b11-96a8-00ebe7da1332.png">

<img width="1792" alt="fte_bte_flowers_2" src="https://user-images.githubusercontent.com/40052470/133530486-28376d7d-a6ce-44a7-ab47-ffa9d45e9ec3.png">

5. fte_bte_food101.ipynb: Mismatch in the input size. May be missing a preprocessing step. 

<img width="1792" alt="fte_bte_food101" src="https://user-images.githubusercontent.com/40052470/133530538-496e29cc-b3fd-4a30-a76b-98fb8ac0611b.png">

6. fte_bte_mnist.ipynb: No errors found. Everything runs smoothly. 

7. label_shuffle_exp.ipynb: 
* Error 1: Convert Keras to tf.keras and import TensorFlow 

<img width="1792" alt="label_shuffle_exp_1" src="https://user-images.githubusercontent.com/40052470/133530618-19403ef5-3891-42b5-a4be-4a00372b5c67.png">

* Error 2: '__init()__ got an unexpected keyword argument 'n_estimators' 

<img width="1792" alt="label_shuffle_exp_2" src="https://user-images.githubusercontent.com/40052470/133530677-761a082c-d7c1-4fbf-a961-6b7551e8a79a.png">

<img width="1792" alt="label_shuffle_exp_3" src="https://user-images.githubusercontent.com/40052470/133530683-6432af27-7a96-4299-af47-05ed94130803.png">

8. random_class_exp.ipynb: Must import TensorFlow and use tf.keras instead of just Keras. Besides this, everything runs smoothly.

9. rotation_cifar: 

* Error 1: Convert Keras to tf.keras and import TensorFlow 
* Error 2: Missing dependency for a background function file

<img width="1792" alt="rotation_cifar" src="https://user-images.githubusercontent.com/40052470/133530957-014c7350-28e8-4555-8af8-ef770708fd31.png">

* Error 3: After rectifying error 1 and 2, the system throws the '__init()__ got an unexpected keyword argument 'n_estimators' error

<img width="1792" alt="rotation_cifar_2" src="https://user-images.githubusercontent.com/40052470/133531036-a9727ddc-2107-448a-9050-559b4c135420.png">

10. sample_size_experiment.ipynb: Error in the background functions file sample_size_functions.py - TensorFlow must be imported and tf.keras used instead. As of now, the code throws the error: module 'keras.optimizers' has no attribute 'Adam'

<img width="1792" alt="sample_size_experiment" src="https://user-images.githubusercontent.com/40052470/133531188-1b1ba270-94e6-428b-b4d9-8d169e3d7d9f.png">

11. spiral_exp.ipynb: No error, but warning issued saying constrained_layout could not be applied. 

<img width="1792" alt="spiral_exp" src="https://user-images.githubusercontent.com/40052470/133531312-4a286146-15fe-41b2-8496-4f19f225142b.png">

12. xor_rxor_exp.ipynb: No error, but warning issued saying constrained_layout could not be applied. 

<img width="1792" alt="xor_rxor_exp" src="https://user-images.githubusercontent.com/40052470/133531621-955706f6-99b0-48cf-853d-ef783e9afb8e.png">

13. xor_rxor_with_cpd.ipynb: 

* Error 1: pycpd is not able to auto-install 
<img width="1792" alt="xor_rxor_with_cpd" src="https://user-images.githubusercontent.com/40052470/133531443-04bfd5a1-ceb0-47da-8f4b-0a1078fe87fb.png">

* Error 2: Even after manually installing via pip and rebuilding the project, the error persisted. 

<img width="1792" alt="rebuilt_proglean" src="https://user-images.githubusercontent.com/40052470/133531477-853b5431-2d7b-4901-ac63-61d62ed02fb9.png">

14. xor_rxor_with_icp.ipynb: No errors found. Everything runs smoothly. 

15. xor_xnor_exp.ipynb: No error, but warning issued saying constrained_layout could not be applied. 

<img width="1792" alt="xor_xnor_exp" src="https://user-images.githubusercontent.com/40052470/133531576-65b655af-3583-40b8-8229-8468d22daab8.png">

---------------------------------------------------------------------------------------------------
SUMMARY:

**Input Mismatch Error**: 
* fte_bte_dtd
* fte_bte_flowers
* fte_bte_food101
 
NOTE: I may be missing some input preprocessing. I visited the TidyLLF link, however that script seems to be running for a lot more datasets than the ones needed for the tutorials, therefore I did not execute it. 

**Unable to Import Dependencies**: 
* ConvRF_Kappas
* ConvRF_Kappas_3
* xor_rxor_with_cpd

NOTE: Not rectified by rebuilding project, restarting kernels, or by restarting Jupyter 

**__init()__ got unexpected keyword argument**:
* label_shuffle_exp
* rotation_cifar

**Tensorflow to be imported and keras changed to tf.keras**
* label_shuffle_exp
* random_class_exp
* rotation_cifar
* sample_size_experiment **Fix Required in Background Function .py File**

**Warnings stating that constrained_layout could not be applied**
* spiral_exp
* xor_rxor_exp
* xor_xnor_exp

---------------------------------------------------------------------------------------------------

**Comment by PSSF23** [comment]
The tutorial/experiment notebooks should be up-to-date now.
